### PR TITLE
For #254: Implementing Images.importFromTar method

### PR DESCRIPTION
**Pull request by bkuzmic**
For #254:

- Renaming importImages to importFromTar (because of the Docker documentation:
  https://docs.docker.com/engine/api/v1.35/#operation/ImageLoad
- Implementing Images.importFromTar method

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#272](https://github.com/amihaiemil/docker-java-api/issues/272) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#272](https://github.com/amihaiemil/docker-java-api/issues/272) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/9f47bb96-893b-400b-b8c3-3684855efbf4); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by paulodamaso** [comment]
@rultor merge

File: src/main/java/com/amihaiemil/docker/Images.java [hunk 1]
```diff
@@ -61,18 +57,15 @@ Image pull(
...
      * @throws IOException If an I/O error occurs.
      * @throws UnexpectedResponseException If the API responds with an
-     *  undexpected status.
+     *  unexpected status.
      */
-    Image importImage(
-        final URL source, final String repo
-    ) throws IOException, UnexpectedResponseException;
+    void importFromTar(
+        String file) throws IOException, UnexpectedResponseException;
```
[the remark this answers is not in the document]
Comment (bkuzmic) [reply src/main/java/com/amihaiemil/docker/Images.java]: @amihaiemil I see why there is a confusion. Create method has an parameter (`fromSrc`) to import image from the source URL and repo (look at: https://docs.docker.com/engine/api/v1.35/#operation/ImageCreate). There is also a separate Import method which uses tar. And the test that was written pointed me to the Import with tar method.
I'll leave the `importImage` intact and create a new puzzle to implement that one as well as create a unit test for it.
Regarding the return type - unfortunately, importing with tar returns empty response body and I don't have a reference to Image name. This is the same for `importImage` with source URL and repo - so I'll change the return type to `void`.

[the remark this answers is not in the document]
Comment (amihaiemil) [reply src/main/java/com/amihaiemil/docker/Images.java]: > Regarding the return type - unfortunately, importing with tar returns empty response body and I don't have a reference to Image name. This is the same for importImage with source URL and repo - so I'll change the return type to void.

@bkuzmic I would say, also leave a puzzle for the return time. We must make it ``Image`` somehow (add new implementations of ``Image``) -- just having ``void`` is useless, the use creates/imports an Image and ... what? Then they have to iterate over the created images and take the latest? Doesn't seem right.

And we should be able to get some kind of Image identifier from somewhere: maybe from the URL or from the tar's filename?

[the remark this answers is not in the document]
Comment (bkuzmic) [reply src/main/java/com/amihaiemil/docker/Images.java]: @amihaiemil Ok, I just did that.

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
The job [#272](https://github.com/amihaiemil/docker-java-api/issues/272) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**